articleview.cc

```diff
@@ -566,54 +566,51 @@ void ArticleView::loadFinished( bool )
   // Expand collapsed article if only one loaded
   ui.definition->page()->mainFrame()->evaluateJavaScript( "gdCheckArticlesNumber();" );
 
-  bool jumpedToCurrentArticle = false;
-  // Jump to current article after page reloading
-  if( !articleToJump.isEmpty() )
+  QVariant userDataVariant = ui.definition->history()->currentItem().userData();
+  if ( userDataVariant.type() == QVariant::Map )
   {
-    jumpedToCurrentArticle = setCurrentArticle( articleToJump, true );
-    articleToJump.clear();
-  }
+    QMap< QString, QVariant > userData = userDataVariant.toMap();
 
-  if( !jumpedToCurrentArticle )
-  {
-    QVariant userDataVariant = ui.definition->history()->currentItem().userData();
+    double sx = 0, sy = 0;
+    bool moveToCurrentArticle = true;
 
-    if ( userDataVariant.type() == QVariant::Map )
+    if ( userData.value( "sx" ).type() == QVariant::Double )
     {
-      QMap< QString, QVariant > userData = userDataVariant.toMap();
-
-      QString currentArticle = userData.value( "currentArticle" ).toString();
-
-      if ( currentArticle.size() )
-      {
-        // There's an active article saved, so set it to be active.
-        setCurrentArticle( currentArticle );
-      }
-
-      double sx = 0, sy = 0;
+      sx = userData.value( "sx" ).toDouble();
+      moveToCurrentArticle = false;
+    }
 
-      if ( userData.value( "sx" ).type() == QVariant::Double )
-        sx = userData.value( "sx" ).toDouble();
+    if ( userData.value( "sy" ).type() == QVariant::Double )
+    {
+      sy = userData.value( "sy" ).toDouble();
+      moveToCurrentArticle = false;
+    }
 
-      if ( userData.value( "sy" ).type() == QVariant::Double )
-        sy = userData.value( "sy" ).toDouble();
+    if ( sx != 0 || sy != 0 )
+    {
+      // Restore scroll position if at least one non-zero coordinate was stored.
+      // Moving to (0, 0) is a no-op, so don't restore it.
+      ui.definition->page()->mainFrame()->evaluateJavaScript(
+          QString( "window.scroll( %1, %2 );" ).arg( sx ).arg( sy ) );
+    }
 
-      if ( sx != 0 || sy != 0 )
-      {
-        // Restore scroll position
-        ui.definition->page()->mainFrame()->evaluateJavaScript(
-            QString( "window.scroll( %1, %2 );" ).arg( sx ).arg( sy ) );
-      }
+    const QString currentArticle = userData.value( "currentArticle" ).toString();
+    if( !currentArticle.isEmpty() )
+    {
+      // There's a current article saved, so set it to be current.
+      // If a scroll position was stored - even (0, 0) - don't move to the
+      // current article.
+      setCurrentArticle( currentArticle, moveToCurrentArticle );
```

Comment:
Reordering `window.scroll` and `setCurrentArticle()` was probably a bad idea. What if a possibly user-customized current article style changes its height? I am going to revert the order to what it was before this change.

Reply:
Done.